From: Developer
Date: Wed, 12 Aug 2026
Subject: [PATCH] stub AudioDeviceCallback for ExoPlayer / NewPipe

NewPipe's ExoPlayer DefaultAudioSink.getAudioCapabilities() references
android.media.AudioDeviceCallback (API 23). ATL had no such class, so
playback failed with ClassNotFoundException / NoClassDefFoundError.

--- a/src/api-impl/android/media/AudioDeviceCallback.java	2026-09-13 17:57:18.322238811 +1000
+++ b/src/api-impl/android/media/AudioDeviceCallback.java	2026-09-13 17:57:18.327238824 +1000
@@ -1,4 +1,9 @@
 package android.media;
 
-public class AudioDeviceCallback {
+public abstract class AudioDeviceCallback {
+	public void onAudioDevicesAdded(AudioDeviceInfo[] addedDevices) {
+	}
+
+	public void onAudioDevicesRemoved(AudioDeviceInfo[] removedDevices) {
+	}
 }
